cdap-master/src/main/java/co/cask/cdap/master/environment/k8s/AppFabricServiceMain.java

```diff
@@ -121,6 +127,35 @@ protected LoggingContext getLoggingContext() {
                                      Constants.Service.APP_FABRIC_HTTP);
   }
 
+  /**
+   * A Guava {@link Service} for creating storages.
+   */
+  private static final class StorageCreationService extends AbstractIdleService {
```

Comment:
I have a PR which does something similar to `StoreDefinition.createAllTables`, but only with a subset of tables, for the Cloud Runtime case: https://github.com/cdapio/cdap/pull/11081
Would it make sense to use the same service there, but passing in a flag?

Reply:
For the one in #11081, I think it is a special workaround, until we removed the reliance on DatasetService entirely. The code change here though, is to stay.

Reply:
Makes sense. So in that PR, I won't try to reuse this class.